DEPRECATION.md

```diff
@@ -0,0 +1,225 @@
+them. Otherwise, you can directly call the `warn` function
+from the [warnings module in the Python standard library](https://docs.python.org/3/library/warnings.html),
+using the category `DeprecationWarning`.  For example:
+
+```
+import warnings
+
+def deprecated_function():
+   warnings.warn(
+      "The function qiskit.deprecated_function() is deprecated since "
+      "Qiskit Terra 0.20.0, and will be removed 3 months or more later. "
+      "Instead, you should use qiskit.other_function().",
+      category=DeprecationWarning,
+      stacklevel=2,
+   )
+   # ... the rest of the function ...
+
+```
+
+Make sure you include the version of the package that introduced the deprecation
+warning (so maintainers can easily see when it is valid to remove it), and what
+the alternative path is.
+
+Take note of the `stackleve`` argument.  This controls which function is
+accused of being deprecated.  Setting `stacklevel=1` (the default) means the
+warning will blame the `warn` function itself, while `stacklevel=2` will
+correctly blame the containing function.  It is unusual to set this to anything
+other than `2`, but can be useful if you use a helper function to issue the
+same warning in multiple places.
+
+
+## Testing deprecated functionality
+
+Whenever you add deprecation warnings, you will need to update tests involving
+the functionality.  The test suite should fail otherwise, because of the new
+warnings.  We must continue to test deprecated functionality throughout the
+deprecation period, to ensure that it still works.
+
+To update the tests, you need to wrap each call of deprecated behavior in its
+own assertion block.  For subclasses of `unittest.TestCase` (which all Qiskit
+test cases are), this is done by:
+
+
+```
+class MyTestSuite(QiskitTestCase):
+   def test_deprecated_function(self):
+      with self.assertWarns(DeprecationWarning):
+         output = deprecated_function()
+      # ... do some things with output ...
+      self.assertEqual(output, expected)
+```
+
+## Documenting deprecations and breaking changes
+
+It is important to warn the user when your breaking changes are coming.
+
+`@deprecate_arg` and `@deprecate_func` will automatically add the deprecation to the docstring
+for the function so that it shows up in docs.
+
+If you are not using those decorators, you should directly add a [Sphinx deprecated directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated):
+
+
+```
+   def deprecated_function():
+      """
+      Short description of the deprecated function.
+
+      .. deprecated:: 0.20.0
+         The function qiskit.deprecated_function() is deprecated since
+         Qiskit Terra 0.20.0, and will be removed 3 months or more later.
+         Instead, you should use qiskit.other_function().
+
+      <rest of the docstring>
+      """
+      # ... the rest of the function ...
+```
```

Comment:
```suggestion
```python
def deprecated_function():
    """
    Short description of the deprecated function.

    .. deprecated:: 0.44.0
       The function qiskit.deprecated_function() is deprecated since
       Qiskit 0.44.0, and will be removed 3 months or more later.
       Instead, you should use qiskit.other_function().

    <rest of the docstring>
    """
    # ... the rest of the function ...
```
```